src/tike/ptycho/probe.py

```diff
@@ -138,6 +138,20 @@ class ProbeOptions:
     hard constraint.
     """
 
+    median_filter_abs_probe: bool = False
+    """Binary switch on whether to apply a median filter to absolute value of 
+    each shared probe mode.
+    """
+
+    median_filter_abs_probe_px: typing.Tuple[float, float] = ( 1.0, 1.0 )
+    """A 2-element tuple with the median filter pixel widths along each dimension.
+    """
+
+    median_filter_abs_probe_period: int = 1
+    """How often (wrt update epochs) we want to apply median filtering to the 
+    absolute value of each shared probe mode.
+    """
```

Comment:
What happens when median_filter_abs_probe_period and update_period are not the same? For example, if you only update the probe every 5th epoch, but you set the filter period to every 3rd period, then you only filter when the epoch is a multiple of both 5 and 3? Is that intentional?

```
update : 0, 5, 10, [15], 20, 25, [30]
filter : 0, 3, 6, 9, 12, [15], 18, 21, 24, 27, [30]
```

Reply:
Yeah, this is kind of a mess...

I'm starting to think we should just have `median_filter_abs_probe` as a binary switch and `median_filter_abs_probe_px` as the median filter pixel widths.

I've removed `median_filter_abs_probe_period` altogether as the hot spot suppression needs this to happen every probe update epoch anyway.

Reply:
Makes sense, thanks!